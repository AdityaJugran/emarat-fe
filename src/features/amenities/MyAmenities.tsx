import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { Row } from 'antd';
import { useApiCall } from 'config/hooks';
import PageTitle from 'features/shared/components/styledComponents/PageTitle';
import Choice from 'features/shared/Choice';
import { AmenityType } from 'features/amenities/Types';
import AmenitySkeleton from 'features/amenities/AmenitySkeleton';
import MyAmenityView from 'features/amenities/MyAmenityView';

export default function MyAmenities() {
	const [choice, setChoice] = useState(0);
	const history = useHistory();

	const { loading: loadingAmenities, data: amenities } = useApiCall({
		apiUrl: 'amenities',
		initDataValue: [],
	});

	const { loading: loadingUserAmenities, data: userAmenities } = useApiCall({
		apiUrl: 'currentUserAmenities',
		initDataValue: [],
	});

	const addAmenity = (id: string, name: string) => {
		history.push(`/complaints?id=${id}&name=${name}`);
	};

	const loading = loadingAmenities || loadingUserAmenities;

	return (
		<>
			<PageTitle className="bg-transparent">
				<Choice
					{...{
						choice,
						setChoice,
						labels: ['My Amenities', 'All Amenities'],
					}}
				/>
			</PageTitle>
			<div className="max-w-screen-lg mx-auto mt-12">
				<Row
					gutter={[
						{ xs: 0, sm: 20 },
						{ xs: 20, sm: 20 },
					]}
					justify="center"
				>
					{loading ? (
						<AmenitySkeleton />
					) : (
						<>
							{amenities
								.filter(
									(amenity: AmenityType) =>
										choice ||
										userAmenities.includes(amenity._id)
								)
								.map((amenity: AmenityType) => (
									<MyAmenityView
										key={amenity._id}
										{...{
											amenity,
											addAmenity,
											userAmenities,
											choice,
										}}
									/>
								))}
						</>
					)}
				</Row>
			</div>
		</>
	);
}
