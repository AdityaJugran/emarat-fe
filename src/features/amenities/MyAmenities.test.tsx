import MyAmenities from 'features/amenities/MyAmenities';
import renderer from 'react-test-renderer';

it('My Amenities', () => {
	const tree = renderer.create(<MyAmenities />).toJSON();
	expect(tree).toMatchSnapshot();
});
