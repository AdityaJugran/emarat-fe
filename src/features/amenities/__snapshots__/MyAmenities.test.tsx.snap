// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Amenities 1`] = `
Array [
  <div
    className="sc-bdfBQB itvOLS bg-transparent"
  >
    <div
      className="ant-radio-group ant-radio-group-solid title-font"
    >
      <label
        className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
      >
        <span
          className="ant-radio-button ant-radio-button-checked"
          style={Object {}}
        >
          <input
            checked={true}
            className="ant-radio-button-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            type="radio"
            value={0}
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          My Amenities
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-button-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            type="radio"
            value={1}
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          All Amenities
        </span>
      </label>
    </div>
  </div>,
  <div
    className="max-w-screen-lg mx-auto mt-12"
  >
    <div
      className="ant-row ant-row-center"
      style={
        Object {
          "marginBottom": -10,
          "marginLeft": -10,
          "marginRight": -10,
          "marginTop": -10,
        }
      }
    />
  </div>,
]
`;
